trio/_util.py

```diff
@@ -122,7 +122,7 @@ def __init__(self, func, args, kwds):
                 # Likewise, avoid suppressing if a StopIteration exception
                 # was passed to throw() and later wrapped into a RuntimeError
                 # (see PEP 479).
-                if exc.__cause__ is value:
+                if type is StopAsyncIteration and exc.__cause__ is value:
```

Comment:
Should check for `StopIteration` as well. I guess `if issubclass(type, (StopIteration, StopAsyncIteration)) and exc.__cause__ is value:` might be the most idiomatic way to write it